src/genpy/message.py

```diff
@@ -65,6 +66,21 @@
 
 struct_I = struct.Struct('<I')
 
+# Notify the user while not crashing in the face of errors attempting
+# to decode non-unicode data within a ROS message.
+class RosMsgUnicodeErrors:
+    def __init__(self):
+        self.data_class = None
+
+    def __call__(self, err):
+        # Lazy import to avoid this cost in the non-error case.
+        import logging
+        logger = logging.getLogger('rosout')
+        extra = "message %s" % self.data_class.__name__ if self.data_class else "unknown message"
+        logger.error("Characters replaced when decoding %s: %s", extra, err)
```

Comment:
Should this be changed to log each kind of message only once to avoid resulting in repeating the same error message (as mentioned in https://github.com/ros/genpy/pull/127#issuecomment-658339090)?

Reply:
AFAICT, log_once is based on the call stack, not the arguments, per: https://docs.ros.org/api/rospy/html/rospy.core-pysrc.html#_frame_to_caller_id

So using a naive `log_once` here would only send the log line once period, not once per kind.

I think someone else will have to pick up any further nits on this; I've already spent far more time on it than it's worth to us.

Reply:
See 12f38689b3983b8d10917ce5506885ee877c4069.